modules/op-triggering-a-pipelinerun.adoc

```diff
@@ -0,0 +1,31 @@
+// This module is included in the following assembly:
+//
+// *openshift_pipelines/op-creating-applications-with-cicd-pipelines.adoc
+
+[id="triggering-a-pipeline_{context}"]
+= Triggering a PipelineRun
+
+Whenever a `push` event happens in the Git repository, the configured Webhook sends an event payload to the configured service route. The EventListener service of the application captures the event and processes the payload using the relevant TriggerBindings and TriggerTemplates. This may re-build and re-deploy the application.
```

Comment:
Whenever user `push` some changes to  Git repository,  configured Webhook sends an event payload to the route (exposed EventListener service). The EventListner will validate and pass the event payload to the TriggerBinding, TriggerTemplate pair, TriggerBinding will extract parameters from payload needed for rendering the TriggerTemplate and creates resources defined under triggersTemplate with extracted data 


@nkakkar81  just parse this if you feel this gives more context

Reply:
Whenever a `push` event happens in the Git repository, the configured Webhook sends an event payload to the publicly exposed EventListener service route. The EventListener service of the application processes the payload, and passes it to relevant TriggerBindings and TriggerTemplates pair. TriggerBinding extracts the parameters and TriggerTemplate uses these parameters to create resources. This may re-build and re-deploy the application.

Reply:
done